packages/salesforcedx-vscode-core/src/commands/forceApexLogGet.ts

```diff
@@ -0,0 +1,204 @@
+    return new SfdxCommandBuilder()
+      .withDescription(nls.localize('force_apex_log_get_text'))
+      .withArg('force:apex:log:get')
+      .withFlag('--logid', data.id)
+      .withArg('--json')
+      .build();
+  }
+
+  protected attachExecution(
+    execution: CommandExecution,
+    cancellationTokenSource: vscode.CancellationTokenSource,
+    cancellationToken: vscode.CancellationToken
+  ) {
+    channelService.streamCommandStartStop(execution);
+    notificationService.reportCommandExecutionStatus(
+      execution,
+      cancellationToken
+    );
+    CancellableStatusBar.show(execution, cancellationTokenSource);
+    taskViewService.addCommandExecution(execution, cancellationTokenSource);
+  }
+
+  public async execute(
+    response: ContinueResponse<ApexDebugLogIdStartTime>
+  ): Promise<void> {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const cancellationToken = cancellationTokenSource.token;
+    const execution = new CliCommandExecutor(this.build(response.data), {
+      cwd: vscode.workspace.rootPath
+    }).execute(cancellationToken);
+    this.attachExecution(execution, cancellationTokenSource, cancellationToken);
+    const result = await new CommandOutput().getCmdResult(execution);
+    const resultJson = JSON.parse(result);
+    if (resultJson.status === 0) {
+      const logDir = path.join(
+        vscode.workspace.workspaceFolders![0].uri.fsPath,
+        '.sfdx',
+        'tools',
+        'debug',
+        'logs'
+      );
+      if (!fs.existsSync(logDir)) {
+        mkdir('-p', logDir);
+      }
+
+      const date = moment(new Date(response.data.startTime)).format(
+        'YYYYMMDDhhmmss'
+      );
+      const logPath = path.join(logDir, `${response.data.id}_${date}.log`);
+      fs.writeFileSync(logPath, resultJson.result.log);
+      const document = await vscode.workspace.openTextDocument(logPath);
+      vscode.window.showTextDocument(document);
+    }
+  }
+}
+
+export type ApexDebugLogIdStartTime = {
+  id: string;
+  startTime: string;
+};
+
+export type ApexDebugLogObject = {
+  Id: string;
+  StartTime: string;
+  LogLength: number;
+  Operation: string;
+  Request: string;
+};
+
+interface ApexDebugLogItem extends vscode.QuickPickItem {
+  id: string;
+}
+
+export class LogFileSelector
+  implements ParametersGatherer<ApexDebugLogIdStartTime> {
+  public async gather(): Promise<
+    CancelResponse | ContinueResponse<ApexDebugLogIdStartTime>
+  > {
+    const cancellationTokenSource = new vscode.CancellationTokenSource();
+    const logInfos = await ForceApexLogList.getLogs(cancellationTokenSource);
+    console.log('loginfos: ' + logInfos);
+    if (logInfos.length > 0) {
+      const logItems = logInfos.map(logInfo => {
+        const icon = '$(file-text) ';
+        return {
+          id: logInfo.Id,
+          label: icon + logInfo.Operation,
+          detail: new Date(logInfo.StartTime).toLocaleTimeString('en-US', {
```

Comment:
Can we use moment here as well? Might as well be consistent and use it throughout.

Reply:
On that note, you should use it in your start logging command as well.